automl/README.md

```diff
@@ -0,0 +1,107 @@
+# AutoML Samples
+
+<a href="https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/golang-samples&page=editor&open_in_editor=automl/README.md">
+<img alt="Open in Cloud Shell" src ="http://gstatic.com/cloudssh/images/open-btn.png"></a>
+
+This directory contains samples for the [Google Cloud AutoML APIs](https://cloud.google.com/automl/) - [docs](https://cloud.google.com/automl/docs/)
+
+We highly reccommend that you refer to the official documentation pages:
+
+* AutoML Natural Language
+  * [Classification](https://cloud.google.com/natural-language/automl/docs)
+  * [Entity Extraction](https://cloud.google.com/natural-language/automl/entity-analysis/docs)
+  * [Sentiment Analysis](https://cloud.google.com/natural-language/automl/sentiment/docs)
+* [AutoML Translation](https://cloud.google.com/translate/automl/docs)
+* AutoML Video Intelligence
+  * [Classification](https://cloud.google.com/video-intelligence/automl/docs)
+  * [Object Tracking](https://cloud.google.com/video-intelligence/automl/object-tracking/docs)
+* AutoML Vision
+  * [Classification](https://cloud.google.com/vision/automl/docs)
+  * [Edge](https://cloud.google.com/vision/automl/docs/edge-quickstart)
+  * [Object Detection](https://cloud.google.com/vision/automl/object-detection/docs)
+* [AutoML Tables](https://cloud.google.com/automl-tables/docs)
+
+This API is part of the larger collection of Cloud Machine Learning APIs.
+
+These Go samples demonstrates how to access the Cloud AutoML API using the
+[Google Cloud Client Library for Go](https://github.com/googleapis/google-cloud-go).
+
+## Sample Types
```

Comment:
For a future PR, so we can get this one in:

Do we need anything after this line? It's very easy to forget to update READMEs and we've generally avoided putting anything that requires updating in them. Pointers to the official docs where these samples are embedded is OK.

Reply:
I was following the Python canonicals, the AutoML samples aren't as straightforward as other samples so I think all languages are including a similar README to explain how they are organized.

Reply:
With the updated filenames, this is now out of date. :/ Already merged, so will you send a followup? I still think we can delete it.

Reply:
Fixing in #1087